src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -116,6 +114,11 @@ def load_arguments(self, _):
         c.argument('platform_fault_domain_count', type=int, help='Fault Domain count.')
         c.argument('validate', help='Generate and validate the ARM template without creating any resources.', action='store_true')
         c.argument('unmanaged', action='store_true', min_api='2016-04-30-preview', help='contained VMs should use unmanaged disks')
+
+    with self.argument_context('vm availability-set update') as c:
+        if self.supported_api_version(max_api='2016-04-30-preview', operation_group='virtual_machines'):
```

Comment:
This rather hacky workaround is needed beause compute changed the parameter name for an availability set from 'name' to 'availability_set_name' between two versions... 🤷‍♂️